abi/valuesSingle.go

```diff
@@ -0,0 +1,106 @@
+package abi
+
+import "math/big"
+
+// U8Value is a wrapper for uint8
+type U8Value struct {
```

Comment:
question: would have been better if we would have avoided these wrapper structs and instead used alias types?

Reply:
Will attempt to do something like that for primitives.

Reply:
For encoding, we could have passed primitive Go values - simply passed by value.

For decoding, we had two choices: either require the caller to pass a (mutable) pointer towards the (empty) primitive placeholder, or to pass a reference of a wrapper.

In JavaScript / Python etc., passing mutable pointers is not an option. And in order to have some consistency for the codecs, across multiple languages, the wrapper-based solution seemed a good option.

However, indeed, we should strive to support primitive values as much as possible, as well. I've added an issue here: https://github.com/multiversx/mx-sdk-abi-go/issues/4.

Is it OK if we postpone this (somehow larger change) to a next PR?

Reply:
Sure, can be postponed. Not a blocker issue